packages/esm-service-queues-app/src/views/queue-tables-for-all-statuses.component.tsx

```diff
@@ -10,23 +10,96 @@ import QueueTable from '../queue-table/queue-table.component';
 }
 
 // displays the queue entries of a given queue by
 // showing one table per status
-const QueueTablesForAllStatuses: React.FC<QueueTablesForAllStatusesProps> = ({ selectedQueue }) => {
+const QueueTablesForAllStatuses: React.FC<QueueTablesForAllStatusesProps> = ({
+  selectedQueue,
+  isLoadingQueue,
+  errorFetchingQueue,
+}) => {
   const layout = useLayoutType();
   const { t } = useTranslation();
 
-  const { queueEntries, isLoading, isValidating } = useQueueEntries({ queue: selectedQueue.uuid, isEnded: false });
-  const allowedStatuses = selectedQueue.allowedStatuses;
   const [searchTerm, setSearchTerm] = useState('');
 
+  if (errorFetchingQueue) {
+    return (
+      <InlineNotification
+        kind="error"
+        title={t('invalidQueue', 'Invalid Queue')}
+        subtitle={errorFetchingQueue?.message}
+      />
+    );
+  }
+
+  return (
+    <>
+      <PatientQueueHeader
+        title={!isLoadingQueue ? selectedQueue?.display : <SkeletonText />}
+        showLocationDropdown={false}
+        actions={
+          <div className={styles.headerButtons}>
+            <ExtensionSlot
+              name="patient-search-button-slot"
+              state={{
+                buttonText: t('addPatientToQueue', 'Add patient to queue'),
+                overlayHeader: t('addPatientToQueue', 'Add patient to queue'),
+                buttonProps: {
+                  kind: 'secondary',
+                  renderIcon: (props) => <Add size={16} {...props} />,
+                  size: isDesktop(layout) ? 'sm' : 'lg',
+                },
+                selectPatientAction: (selectedPatientUuid) => {
+                  launchWorkspace('service-queues-patient-search', {
+                    selectedPatientUuid,
+                    currentServiceQueueUuid: selectedQueue.uuid,
+                  });
+                },
+              }}
+            />
+            <div className={styles.filterSearch}>
+              <Search
+                labelText=""
+                placeholder={t('filterTable', 'Filter table')}
+                onChange={(e) => setSearchTerm(e.target.value)}
+                size={isDesktop(layout) ? 'sm' : 'lg'}
+                disabled={isLoadingQueue}
+              />
+            </div>
+          </div>
+        }
+      />
+      {!isLoadingQueue && <QueueTableMetrics selectedQueue={selectedQueue} />}
+      {!isLoadingQueue ? (
+        <QueueTablesByStatus selectedQueue={selectedQueue} searchTerm={searchTerm} />
+      ) : (
+        <QueueTableByStatusSkeleton />
+      )}
+    </>
+  );
+};
+
+interface QueueTablesByStatusProps {
+  selectedQueue: Queue;
+  searchTerm: string;
+}
+
+function QueueTablesByStatus({ selectedQueue, searchTerm }: QueueTablesByStatusProps) {
+  const { t } = useTranslation();
+  const { queueEntries, isLoading, isValidating } = useQueueEntries({ queue: selectedQueue.uuid, isEnded: false });
+  const allowedStatuses = selectedQueue.allowedStatuses.reverse();
```

Comment:
This is being reversed again below, so this (or that) should not be reversed.

Reply:
Thanks for the catch @mseaton!
Pushed the change.
Thanks!